src/index.js

```diff
@@ -32,87 +34,148 @@ const getCommands = getQueryNames.map(queryName => {
+        // make the timeout extremely short to ensure `query*` commands pass or fail instantly
+        options.timeout = 0
+      }
+
+      let error
+      let failedNewFunctionality = false
+      let failedOldFunctionality = false
+      // Errors will be thrown by @testing-library/dom, but a query might be followed by `.should('not.exist')`
+      // We just need to capture the error thrown by @testing-library/dom and return an empty jQuery NodeList
+      // to allow Cypress assertions errors to happen naturally. If an assertion fails, we'll have a helpful
+      // error message handy to pass on to the user
+      const catchQueryError = err => {
+        error = err
+        failedOldFunctionality = true
+        const result = Cypress.$()
+        result.selector = getSelector()
+        return result
+      }
+
+      // Before https://github.com/testing-library/cypress-testing-library/pull/100,
+      // queries were run without being scoped to previous subjects. There is code now that depends
+      // on functionality before #100. See if we can succeed using old functionality before finally failing
+      // This function can be removed as a breaking change
+      const catchAndTryOldFunctionality = err => {
+        error = err
+        failedNewFunctionality = true
+        const container = options.fallbackToPreviousFunctionality ? options.container || win.document : undefined
+        return getValue(container)
+      }
 
-            return subject
-          })
+      const resolveValue = () => {
+        // retry calling "getValue" until following assertions pass or this command times out
+        return Cypress.Promise.try(getValue)
+          .catch(catchAndTryOldFunctionality)
+          .catch(catchQueryError)
+          .then(value => {
+            return cy.verifyUpcomingAssertions(value, options, {
+              onRetry: resolveValue,
+              onFail: () => {
+                // We want to override Cypress's normal non-existence message with @testing-library/dom's more helpful ones
+                if (error) {
+                  options.error = error
+                }
+              }
+            })
         })
+      }
+
+      return resolveValue().then(subject => {
+        // Remove the error that occurred because it is irrelevant now
+        if (consoleProps.error) {
+          delete consoleProps.error
+        }
+        if (options._log) {
+          options._log.snapshot()
+        }
+
+        return subject
+      }).finally(() => {
+        if (options._log) {
+          if (failedNewFunctionality && !failedOldFunctionality) {
+            options._log.error(Error(`@testing-library/cypress will soon only use previous subjects when queries are added to a chain of commands. We've detected an instance where the this functionality failed, but the old functionality passed. Please use cy.${queryName}(${queryArgument(args)}) instead of continuing from a previous chain.`))
```

Comment:
Cypress doesn't have good deprecation or warning message primitives. This code sets the command in "error" mode and attaches the error on the log, but doesn't show any warning message in the command log. What do people think of this? This triggers when backwards compatibility mode usage was detected prompting an upgrade. Is the message enough?

<img width="928" alt="Screen Shot 2020-01-30 at 10 54 01 PM" src="https://user-images.githubusercontent.com/338257/73515702-7cc95e00-43b3-11ea-8873-806dd1bc81b4.png">

Reply:
Wish it didn't say "Error" and instead said "Warning" but that's ok. Could we change the message slightly?

```
@testing-library/cypress will eventually only use previous subjects when queries are added to a chain of commands. We've detected an instance where the this functionality failed, but the old functionality passed (so your test may break in a future version). Please use cy.${queryName}(${queryArgument(args)}) instead of continuing from a previous chain.
```

Reply:
> Wish it didn't say "Error" and instead said "Warning"

Agreed. Perhaps that could be a feature request for Cypress to allow for us to trigger warnings